_rules/disable-shortcut-670a30.md

```diff
@@ -0,0 +1,326 @@
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters, a mechanism to disable the shortcut exists.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [HTML document][] with at least one [keyboard shortcut][] that requires pressing only [printable character][] keys.
+
+## Expectation
+
+For the test target if any [keyboard shortcut][] requires pressing only [printable character][] keys, a [mechanism][] to disable the shortcut exists.
+
+If the [mechanism][] to disable the shortcut is a [user interface component][], then it must be [visible][] and [included in the accessibility tree][] with an [accessible name][] that is not empty (`""`).
+
+## Assumptions
+
+This rule assumes as applicable [keyboard shortcuts][keyboard shortcut] those implemented by the test target [content][]. Any other means (e.g. browser extensions, browser settings, user agents, external browser applications) are not considered.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.1.4: Character Key Shortcuts](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html)
+- [G127 Providing a mechanism to allow users to remap or turn off character key shortcuts](https://www.w3.org/WAI/WCAG21/Techniques/general/G217)
+- [F99 Failure of Success Criterion 2.1.4 due to implementing character key shortcuts that cannot be turned off or remapped](https://www.w3.org/WAI/WCAG21/Techniques/failures/F99)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The [HTML document][] has a [keyboard shortcut][] that uses one [printable][printable character] and one [non-printable characters][].
+
+```html
+<html>
+  <head>
+    <title>Passed Example 1 for rule 670a30</title>
+    <script>
+      function shortcut() {
+        document.body.addEventListener('keydown', function(event) {
+          if (event.key === 'i' && event.ctrlKey) {
+            const text = document.getElementById('text');
+            text.className = text.className === 'italic' ? '' : 'italic';
+          }
+        });
+      }
+    </script>
+    <style>
+      .italic {
+        font-style: italic;
+      }
+    </style>
+  </head>
+  <body onload="shortcut();">
+    <div>Press <strong>ctrl+i</strong> to toggle italic format</div>
+    <div id="text">Some text inside the document content</div>
+  </body>
+</html>
+```
+
+#### Passed Example 2
+
+The [HTML document][] has a [keyboard shortcut][] using only a [printable character][], and it can be disabled.
```

Comment:
Here the mechanism to disable is a UI component (checkbox).
Do you have an example where the mechanism to disable is not a UI component? (to complete coverage of the expectation)

Reply:
I'm not feeling creative... if you have a suggestion for a non-widget component to do it, it would be appreciated

Reply:
Hum…
A bit difficult with the WCAG definition of UI component… Which is a bit of the point 😉 

If it is not possible to have the disabling mechanism **not** being a UI component (because, it's always going to be "perceived by users as a single control for a distinct function"), then the expectation should reflect that and read "The UI component to disable …" instead of "If the disabling is done by a UI component …"
(maybe with an assumption that the disabling thingie is perceived as a single control and thus is a UI component).

Reply:
Coming back to that…
A case where disabling is not done by a UI component could be (I guess) if it's done by a global key shortcut (or other kind of global event) not linked to a specific element on the page.

e.g. Pressing `ctrl -` is intercepted by a global window event and disable (or, for other rules, remap) the shortcut. That would not be a UI component (I think) but would still be a (WCAG) "mechanism" so it should still be good for WCAG.
Do we want to allow that? There's a risk of getting into somewhat ambiguous definitions around "mechanism", so maybe we can stick with "disabled by a UI component" for now…
In any case, we need a corresponding passed or failed example with this situation (the disabling/remapping/… mechanism is not a UI component)